### PR TITLE
strtol: Add parse/consume for string_view friendly interface

### PR DESCRIPTION
**Pull request by adamemerson**
With fallback for older Ubuntu and other systems lacking <charconv>.

### CODE REVIEW

File: src/common/strtol.h [hunk 1]
```diff
@@ -15,11 +15,124 @@
 #ifndef CEPH_COMMON_STRTOL_H
 #define CEPH_COMMON_STRTOL_H
 
+#if __has_include(<charconv>)
+#include <charconv>
+#endif // __has_include(<charconv>)
```
[the remark this answers is not in the document]
Comment (cbodley) [reply src/common/strtol.h:20]: i remember @ofriedma was looking at this - did you make any progress there?

[the remark this answers is not in the document]
Comment (adamemerson) [reply src/common/strtol.h:20]: I assume it's just a property of whatever version of the libstdc++ some of our gitbuilders have. charconv took a while to get supported and still doesn't support floats.

[the remark this answers is not in the document]
Comment (ofriedma) [reply src/common/strtol.h:20]: Currently the problem occurs on Ubuntu Bionic(18.04) and Opensuse.
When trying to compile on Bionic as a container it worked well(the compiler found charconv).
Im trying to figure out why it happens on shaman and not on local running.
It doesnt happen anymore in Centos (8)

[the remark this answers is not in the document]
Comment (adamemerson) [reply src/common/strtol.h:20]: I'm not sure that having the fallback really hurts us, and we aren't blocked behind Shaman Stuff.

**Comment by adamemerson** [comment]
This got pulled in with wip-fifo-datalog.

**Pull request closed without merging**